src/ObjCRuntime/Blocks.cs

```diff
@@ -83,14 +83,19 @@ public unsafe struct BlockLiteral {
 		[DllImport ("__Internal")]
 		static extern IntPtr xamarin_get_block_descriptor ();
 
+		struct DelegatePairs {
+			public Delegate Trampoline, UserDelegate;
+		}
+
 		//
 		// trampoline must be static, and someone else needs to keep a ref to it
 		//
 		public unsafe void SetupBlock (Delegate trampoline, Delegate userDelegate)
 		{
 			isa = block_class;
 			invoke = Marshal.GetFunctionPointerForDelegate (trampoline);
-			local_handle = (IntPtr) GCHandle.Alloc (userDelegate);
+			var delegates = new DelegatePairs () { Trampoline = trampoline, UserDelegate = userDelegate };
```

Comment:
It's extra memory that will be used every time a block is set up, when it's not needed for 99.99% of the time since the generated code does the right thing.

Also, we decide to do this change, a Tuple can be used (avoids the need for a new class), and the code that fetches local_handle would have to be updated to cope with the fact that the stored object is different: https://github.com/migueldeicaza/xamarin-macios/blob/7c5a2de2be81094e5597ce3ef447f6a34d24d6a2/src/ObjCRuntime/Blocks.cs#L160